src/main/java/org/agbrothers/englishbot/processing/ProcessingCore.java

```diff
@@ -31,6 +31,7 @@ public void processUserRequest(String chatId, String messageText) {
         User user = userService.getUserByChatId(chatId);
         if (null == user) {
             user = new User(chatId);
+            userService.saveAndFlushUser(user);
```

Comment:
to be sure, you should use the reference to the object, that was added to persistence context, e.g. `user =  userService.saveAndFlushUser(...`